File: app/src/main/java/ru/evka/takeandremember/adapter/TodoListAdapter.java
package ru.evka.takeandremember.adapter;

import android.util.Log;
import android.view.*;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import org.jetbrains.annotations.NotNull;
import ru.evka.takeandremember.R;

import java.util.List;

public class TodoListAdapter extends RecyclerView.Adapter<TodoListItemHolder>
{
    private List<String> items;

    public TodoListAdapter(List<String> items)
    {
        this.items = items;
    }

    @NonNull
    @NotNull
    @Override
    public TodoListItemHolder onCreateViewHolder(@NonNull @NotNull ViewGroup parent, int viewType)
    {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        View view = inflater.inflate(R.layout.list_item, parent, false);
        return new TodoListItemHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull @NotNull TodoListItemHolder holder, int position)
    {
        holder.itemView.<TextView>findViewById(R.id.item_title).setText(items.get(position));
    }

    @Override
    public int getItemCount()
    {
        return items.size();
    }
}

File: app/src/main/java/ru/evka/takeandremember/adapter/TodoListAdapter.kt
package ru.evka.takeandremember.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import ru.evka.takeandremember.R

class TodoListAdapter(private val items: List<String>) : RecyclerView.Adapter<TodoListItemHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TodoListItemHolder {
        val inflater = LayoutInflater.from(parent.context)
        val view: View = inflater.inflate(R.layout.list_item, parent, false)
        return TodoListItemHolder(view)
    }

    override fun onBindViewHolder(holder: TodoListItemHolder, position: Int) {
        holder.itemView.findViewById<TextView>(R.id.item_title).text = items[position]
    }

    override fun getItemCount(): Int {
        return items.size
    }
}